src/test/scala/freeeventsourcing/eventselector/AggregateEventSelectorTests.scala

```diff
@@ -0,0 +1,109 @@
+package freeeventsourcing.eventselector
+
+import freeeventsourcing.EventSelector.WithEventType
+import freeeventsourcing.EventSelector.ops._
+import freeeventsourcing.EventTime.Zero
+import freeeventsourcing.accountprocessing.Account.Event.{ Closed, Opened }
+import freeeventsourcing.accountprocessing.Transaction.Event.Created
+import freeeventsourcing.accountprocessing._
+import freeeventsourcing.utils.CompositeName
+import freeeventsourcing._
+import org.scalatest.{ FlatSpec, Matchers }
+import shapeless.{ ::, HNil }
+
+class AggregateEventSelectorTests extends FlatSpec with Matchers {
+  val router = AggregateEventSelector.Router.forAggregateType(Account)
+
+  def mockMetadata = EventMetadata(MockEventId(), MockEventTime())
+  val openedEvent = AggregateEvent[Account.type, Opened](Account, Account.Id(1), Opened("Mario"))
+
+  "AggregateEventSelector.Router " should " produce the same topic as the matching AggregateEventSelector" in {
+    val toIndex = router.apply(EventWithMetadata(openedEvent, mockMetadata))
+    toIndex.size shouldBe 1
+    val indexed = toIndex.head
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/msiegenthaler/free-event-sourcing/file/2674073933/issues/source?bid=3298199&fileBranchId=3311368#l23)